provider/galaxy.go

```diff
@@ -57,16 +57,21 @@ func (g AnsibleGalaxy) VersionsForRole(ctx context.Context, r requirements.Role)
 		keywords = r.Name
 	}
 
+	// set the Ansible Galaxy search parameters
+	params := url.Values{}
+
 	// namespace to be used to filter the Ansible Galaxy results
 	var namespace string
 	split := strings.Split(keywords, ".")
 	if len(split) > 0 {
 		namespace = split[0]
+		params.Add("namespaces", namespace)
+		params.Add("keywords", split[1])
+	} else {
+		params.Add("keywords", keywords)
 	}
 
-	// set the Ansible Galaxy search parameters
-	params := url.Values{}
-	params.Add("keywords", keywords)
+	params.Add("order_by", "-relevance")
```

Comment:
If I am not mistaken, we don't need anymore the namespace variable now.
We can then rewrite this to

```suggestion
	params := url.Values{} 
	params.Add("order_by", "-relevance")
 
	split := strings.Split(keywords, ".")
	if len(split) > 0 {
		params.Add("namespaces", split[0])
		params.Add("keywords", split[1])
	} else {
		params.Add("keywords", keywords)
	}
```